media/transcoder/transcoder_test.go

```diff
@@ -324,6 +327,80 @@ func testJobFromPreset(t *testing.T, projectNumber string, inputURI string, outp
 	})
 }
 
+// testJobFromPresetBatchMode tests major operations on a job created from a
+// preset in batch mode. It will wait until the job successfully completes as
+// part of the test.
+func testJobFromPresetBatchMode(t *testing.T, projectNumber string, inputURI string, outputURIForPresetBatchMode string) {
+	tc := testutil.SystemTest(t)
+	buf := &bytes.Buffer{}
+	jobID := ""
+
+	// Create the job.
+	jobName := fmt.Sprintf("projects/%s/locations/%s/jobs/", projectNumber, location)
+	if err := createJobFromPresetBatchMode(buf, tc.ProjectID, location, inputURI, outputURIForPresetBatchMode, preset); err != nil {
+		t.Errorf("createJobFromPresetBatchMode got err: %v", err)
+	}
+	got := buf.String()
+
+	if !strings.Contains(got, jobName) {
+		t.Errorf("createJobFromPresetBatchMode got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, jobName)
+	}
+	strSlice := strings.Split(got, "/")
+	jobID = strSlice[len(strSlice)-1]
+	buf.Reset()
+
+	// Get the job by job ID.
+	testutil.Retry(t, 3, 5*time.Second, func(r *testutil.R) {
+		jobName := fmt.Sprintf("projects/%s/locations/%s/jobs/%s", projectNumber, location, jobID)
+		if err := getJob(buf, tc.ProjectID, location, jobID); err != nil {
+			r.Errorf("getJob got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, jobName) {
+			r.Errorf("getJob got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, jobName)
+		}
+	})
+	buf.Reset()
+
+	// Get the job state, which should be succeeded. If the job is still running on the last attempt, pass the test.
+	testutil.Retry(t, 3, 30*time.Second, func(r *testutil.R) {
```

Comment:
Are these timings confirmed with the API team that the job is expected to complete within these bounds?

Reply:
These test jobs were intentionally made to complete quickly and have been running for about 2 years now.

Reply:
Sounds good, just wanted to check.